### ISSUE TITLE
[BUG] TicTacToeEnv.win is not batched and rand_action(None) crashes

### ISSUE THREAD

**Issue by YeonwooSung** [pytorch/rl#4342]
## Describe the bug

`TicTacToeEnv` documents batched resets (`env.reset(TensorDict(batch_size=...))`), but `win()` uses Python `if tensor.sum() == 3`. That reduces over the whole batch, so one env completing a line can mark every env done, or a real win can be missed if the global sum is not 3.

`rand_action` also assumes a tensordict is passed and does `tensordict.get("mask")`. `EnvBase.rand_action(None)` is the public contract (same class of bug as #4225 on ChessEnv). `TicTacToeEnv().rand_action()` raises `AttributeError`.

## To Reproduce

```python
import torch
from tensordict import TensorDict
from torchrl.envs.custom.tictactoeenv import TicTacToeEnv

env = TicTacToeEnv()
env.rand_action()  # AttributeError

# Batched win: only env 0 has a completed row after the last action
```

## Expected behavior

- `win()` returns a per-env bool tensor (shape matching `turn` / `done`).
- `rand_action()` without a tensordict returns a TensorDict with `"action"`.
- With a reset tensordict, `rand_action` still honors `"mask"`.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the documentation (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)
